packages/text-to-speech/src/index.ts

```diff
@@ -85,301 +56,13 @@ const updateSpeech = async (
-    emphasisElement.setAttribute('level', level)
-    emphasisElement.innerHTML = element.innerHTML.trim()
-    parent.appendChild(emphasisElement)
-    replaceElement(parent, element)
-  }
-
-  const replaceWithSentence = (element: Element) => {
-    const parent = ssml.createDocumentFragment() as unknown as Element
-    appendBookmarkElement(parent, element)
-    const sentenceElement = ssml.createElement('s')
-    sentenceElement.innerHTML = element.innerHTML.trim()
-    parent.appendChild(sentenceElement)
-    replaceElement(parent, element)
-  }
-
-  // create new ssml document
-  const ssml = parseHTML('').document
-  const speakElement = ssml.createElement('speak')
-  speakElement.setAttribute('version', '1.0')
-  speakElement.setAttribute('xmlns', 'http://www.w3.org/2001/10/synthesis')
-  speakElement.setAttribute('xml:lang', language)
-  const voiceElement = ssml.createElement('voice')
-  voiceElement.setAttribute('name', voice)
-  speakElement.appendChild(voiceElement)
-  const prosodyElement = ssml.createElement('prosody')
-  prosodyElement.setAttribute('rate', `${rate}`)
-  prosodyElement.setAttribute('volume', volume.toString())
-  voiceElement.appendChild(prosodyElement)
-  // add each paragraph to the ssml document
-  appendBookmarkElement(prosodyElement, htmlElement)
-  // replace emphasis elements with ssml
-  htmlElement.querySelectorAll('*').forEach((e) => {
-    switch (e.tagName.toLowerCase()) {
-      case 's':
-        replaceWithEmphasis(e, 'moderate')
-        break
-      case 'sub':
-        if (e.getAttribute('alias') === null) {
-          replaceWithEmphasis(e, 'moderate')
-        }
-        break
-      case 'i':
-      case 'em':
-      case 'q':
-      case 'blockquote':
-      case 'cite':
-      case 'del':
-      case 'strike':
-      case 'sup':
-      case 'summary':
-      case 'caption':
-      case 'figcaption':
-        replaceWithEmphasis(e, 'moderate')
-        break
-      case 'b':
-      case 'strong':
-      case 'dt':
-      case 'dfn':
-      case 'u':
-      case 'mark':
-      case 'th':
-      case 'title':
-      case 'var':
-        replaceWithEmphasis(e, 'moderate')
-        break
-      case 'li':
-        replaceWithSentence(e)
-        break
-      default: {
-        const parent = ssml.createDocumentFragment() as unknown as Element
-        appendBookmarkElement(parent, e)
-        const text = (e as HTMLElement).innerText.trim()
-        const textElement = ssml.createTextNode(text)
-        parent.appendChild(textElement)
-        replaceElement(parent, e)
-      }
-    }
-  })
-  prosodyElement.appendChild(htmlElement)
-
-  return speakElement.outerHTML.replace(/&nbsp;|\n/g, '')
-}
-
 export const textToSpeechHandler = Sentry.GCPFunction.wrapHttpFunction(
   async (req, res) => {
-    console.debug('New text to speech request', req)
+    console.info('Text to speech request received')
     const token = req.query.token as string
```

Comment:
Could we pull this from the `Authorization` header the same way we do in API?

I think it would be nice if you can supply either way, as URL param or in `Authorization` header.

Reply:
Yes, good advice!